### PR TITLE
feat(maestro): shared WalletConnect Pay E2E test actions

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary

- Add three reusable composite actions for Maestro Pay E2E testing across wallet platforms (React Native, Kotlin, Swift, Flutter)
  - `maestro/setup` — Install Maestro CLI
  - `maestro/pay-tests` — Copy 11 test flows + 3 sub-flows + 2 helper scripts into consumer workspace
  - `maestro/run` — Execute tests with WalletConnect Pay secret passthrough + artifact upload
- Add comprehensive README documenting all 28 required testIDs, 6 merchant secrets, test input field requirement (with RN reference implementation), test catalog, and iOS/Android usage examples

## Context

Built from the Maestro E2E tests developed in [react-native-examples](https://github.com/reown-com/react-native-examples) across PRs #452, #455, #457, #458. These tests cover the full WalletConnect Pay lifecycle: happy paths (single/multi option, KYC/no-KYC), error handling (insufficient funds, expired, cancelled), navigation (back/close buttons), edge cases (double scan, deep links), and server-side cancellation scenarios.

## Test plan

- [x] Point a wallet sample's CI workflow at `WalletConnect/actions/maestro/setup@feat/maestro-pay-actions`, `WalletConnect/actions/maestro/pay-tests@feat/maestro-pay-actions`, and `WalletConnect/actions/maestro/run@feat/maestro-pay-actions`
- [x] Trigger E2E workflow on iOS — confirm flows are copied, Maestro installs, tests run, artifacts upload
- [x] Trigger E2E workflow on Android — confirm inline `maestro test` command works with copied flows
- [x] Verify README renders correctly on GitHub

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by ignaciosantise** [comment]
@claude review

File: maestro/setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          echo "Maestro already installed: $(maestro --version)"
+          exit 0
+        fi
+
+        if [ -n "${{ inputs.version }}" ]; then
+          export MAESTRO_VERSION="${{ inputs.version }}"
+        fi
+
+        curl -fsSL "https://get.maestro.mobile.dev" | bash
+        echo "$HOME/.maestro/bin" >> "$GITHUB_PATH"
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply maestro/setup/action.yml:27]: Fixed — added `set -euo pipefail` at the top of the run block so a failed install properly fails the step.

File: maestro/pay-tests/action.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        ACTION_DIR="${{ github.action_path }}"
+        TARGET_DIR="${{ github.workspace }}/${{ inputs.target-dir }}"
+
+        mkdir -p "$TARGET_DIR/flows" "$TARGET_DIR/scripts"
+
+        # Copy root-level pay test flows
+        cp "$ACTION_DIR"/.maestro/pay_*.yaml "$TARGET_DIR/"
+
+        # Copy shared sub-flows
+        cp "$ACTION_DIR"/.maestro/flows/pay_*.yaml "$TARGET_DIR/flows/"
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply maestro/pay-tests/action.yml:32]: Fixed — added `set -euo pipefail` at the top of the copy script.

File: maestro/pay-tests/.maestro/scripts/create-payment.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Creates a WalletConnect Pay payment via the API.
+// Expects WPAY_CUSTOMER_KEY and WPAY_MERCHANT_ID env vars from Maestro.
+// Sets output.gateway_url and output.payment_id for use in subsequent flow steps.
+
+var response = http.post('https://api.pay.walletconnect.com/v1/payments', {
+  headers: {
+    'Content-Type': 'application/json',
+    'Api-Key': WPAY_CUSTOMER_KEY,
+    'Merchant-Id': WPAY_MERCHANT_ID,
+  },
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply maestro/pay-tests/.maestro/scripts/create-payment.js:13]: Fixed — added explicit `typeof` checks for `WPAY_CUSTOMER_KEY` and `WPAY_MERCHANT_ID`, matching the pattern already used in `cancel-payment.js`.

File: maestro/pay-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+          .maestro/
+
+  - name: Upload Maestro artifacts
+    if: always()
+    uses: actions/upload-artifact@v4
+    with:
+      name: maestro-android-artifacts
+      path: |
+        maestro-artifacts/
+        maestro-output.log
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply maestro/pay-tests/README.md]: Fixed — removed `maestro-output.log` from the artifact upload paths since the inline script doesn't produce it.

File: maestro/pay-tests/.maestro/pay_cancelled.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+name: WalletConnect Pay - Cancelled Payment
+tags:
+  - pay
+---
+# Use a hardcoded cancelled payment URL (no API call needed)
+- evalScript: ${output.gateway_url = 'https://pay.walletconnect.com/?pid=pay_24a2ecc101KNHMZ1B8GJMR16VXAZ2EXXMN'}
+
+- startRecording: "WalletConnect Pay Cancelled Payment"
+
+# Open wallet, paste payment URL
```
[the remark this answers is not in the document]
Comment (ignaciosantise) [reply maestro/pay-tests/.maestro/pay_cancelled.yaml]: Intentionally keeping the hardcoded URL. This flow tests the "open a known-cancelled payment" scenario without API round-trips — same pattern as `pay_expired_link.yaml`. The dynamic create→cancel→verify path is already covered by `pay_cancel_from_review` and `pay_cancel_from_kyc` flows.

**Review by jakubuid — APPROVED** [review]
Love this!

File: maestro/run/action.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  maestro-dir:
+    description: 'Directory containing Maestro test flows'
+    required: false
+    default: '.maestro'
+  tags:
+    description: 'Maestro --include-tags value (comma-separated)'
+    required: false
+    default: 'pay'
+  wpay-customer-key-single-nokyc:
+    description: 'WalletConnect Pay API key (single-option, no-KYC merchant)'
```
Comment (ignaciosantise) [inline maestro/run/action.yml]:
```suggestion
    description: 'WalletConnect Pay Customer API key (single-offramp-option, no-KYC)'
```

File: maestro/run/action.yml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    required: true
+  wpay-merchant-id-single-nokyc:
+    description: 'WalletConnect Pay merchant ID (single-option, no-KYC merchant)'
```
Comment (ignaciosantise) [inline maestro/run/action.yml]:
```suggestion
    description: 'WalletConnect Pay merchant ID (single-offramp-option, no-KYC)'
```

File: maestro/run/action.yml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    required: true
+  wpay-customer-key-multi-nokyc:
+    description: 'WalletConnect Pay API key (multi-option, no-KYC merchant)'
```
Comment (ignaciosantise) [inline maestro/run/action.yml]:
```suggestion
    description: 'WalletConnect Pay Customer API key (multiple-offramp-options, no-KYC)'
```

File: maestro/run/action.yml [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    required: true
+  wpay-merchant-id-multi-nokyc:
+    description: 'WalletConnect Pay merchant ID (multi-option, no-KYC merchant)'
```
Comment (ignaciosantise) [inline maestro/run/action.yml]:
```suggestion
    description: 'WalletConnect Pay merchant ID (multiple-offramp-options, no-KYC))'
```

File: maestro/run/action.yml [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    required: true
+  wpay-customer-key-multi-kyc:
+    description: 'WalletConnect Pay API key (multi-option, KYC-required merchant)'
```
Comment (ignaciosantise) [inline maestro/run/action.yml]:
```suggestion
    description: 'WalletConnect Pay Customer API key (multiple-offramp-options, KYC-required)'
```

File: maestro/run/action.yml [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    required: true
+  wpay-merchant-id-multi-kyc:
+    description: 'WalletConnect Pay merchant ID (multi-option, KYC-required merchant)'
```
Comment (ignaciosantise) [inline maestro/run/action.yml]:
```suggestion
    description: 'WalletConnect Pay merchant ID (multiple-offramp-options, KYC-required)'
```
